components/reddit/sources/new-comments-on-a-subreddit/new-comments-on-a-subreddit.js

```diff
@@ -0,0 +1,113 @@
+const reddit = require("../../reddit.app.js");
+module.exports = {
+  key: "new-comments-on-a-subreddit",
+  name: "New comments on a subreddit",
+  description:
+    "Emits an event each time a new comment is added to a subreddit.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      propDefinition: [reddit, "subreddit"],
+    },
+    subreddit_id36: {
+      type: "string",
+      label: "ID36 of Subreddit",
+      description:
+        "Pipedream will emit a new event when a comment is posted to the subreddit pointed by this ID36.",
+      optional: false,
+    },
+    context: {
+      type: "integer",
+      label: "Context (0 to 8)",
+      description:
+        "If set to 0, it will include the new comment in the emitted event. Otherwise, it specifies how many parents of the new comment will be included in said event.",
+      default: 0,
+      optional: false,
+    },
+    depth: {
+      type: "integer",
+      label: "Depth",
+      description:
+        'If set to 1, it will include, in the emitted event, only new comments that are direct children to the subreddit pointed by "subreddit_id36". Furthermore, "depth" determines the maximum depth of children, within the related subreddit comment tree, of new comments to be included in said emitted event.',
+      default: 1,
+      optional: true,
+    },
+    sr_detail: {
```

Comment:
It looks like at least this prop is used across multiple components - is that right? I'd recommend moving all of these props (besides the `timer`) to the app file, and including them here.

Reply:
The props `sr_detail` and `t` are reused, so those should be setup in the app file under propDefinitions and included just like the `subreddit` prop. As Dylan recommended, you can set all the props up in the app file and include them in the appropriate components. The `timer` prop could be put in a common.js file.

Reply:
Yeah I went ahead and mover `sr_detail`, `t` to the app file. Abstracted out `db`,`timer` props, helper method to the common file, as well as the event emitter method.